internal/services/gateway/action/invitation.go

```diff
@@ -0,0 +1,265 @@
+}
+
+func (h *ActionHandler) OrgInvitationAction(ctx context.Context, req *OrgInvitationActionRequest) error {
+	if !req.Action.IsValid() {
+		return errors.Errorf("action is not valid")
+	}
+
+	userID := common.CurrentUserID(ctx)
+	if userID == "" {
+		return util.NewAPIError(util.ErrBadRequest, errors.Errorf("user not authenticated"))
+	}
+
+	orgInvitation, _, err := h.configstoreClient.GetOrgInvitation(ctx, req.OrgRef, req.UserRef)
+	if err != nil {
+		return util.NewAPIError(util.KindFromRemoteError(err), errors.Wrapf(err, "failed to get org invitation"))
+	}
+	if orgInvitation == nil {
+		return util.NewAPIError(util.ErrBadRequest, errors.Errorf("invitation for org %s user %s not found", req.OrgRef, req.UserRef))
+	}
+
+	if userID != orgInvitation.UserID {
+		return errors.Errorf("user not authorized")
+	}
+
+	org, err := h.GetOrg(ctx, req.OrgRef)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get org %s:", req.OrgRef)
+	}
+	if org == nil {
+		return util.NewAPIError(util.ErrBadRequest, errors.Errorf("org %s not found", req.OrgRef))
+	}
+
+	switch req.Action {
+	case types.Accept:
+		err = h.acceptOrgInvitation(ctx, *orgInvitation)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	case types.Reject:
+		if _, err = h.configstoreClient.DeleteOrgInvitation(ctx, req.OrgRef, req.UserRef); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	return nil
+}
+
+func (h *ActionHandler) DeleteOrgInvitation(ctx context.Context, orgRef string, userRef string) error {
+	userID := common.CurrentUserID(ctx)
+	if userID == "" {
+		return errors.Errorf("user not authenticated")
+	}
+
+	orgInvitation, _, err := h.configstoreClient.GetOrgInvitation(ctx, orgRef, userRef)
+	if err != nil {
+		return util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+
+	if userID != orgInvitation.UserID {
+		org, err := h.GetOrg(ctx, orgInvitation.OrganizationID)
+		if err != nil {
+			return err
+		}
+
+		isOrgOwner, err := h.IsOrgOwner(ctx, org.ID)
+		if err != nil {
+			return util.NewAPIError(util.ErrBadRequest, errors.Wrapf(err, "failed to determine ownership"))
+		}
+		if !isOrgOwner {
+			return util.NewAPIError(util.ErrForbidden, errors.Errorf("user is not owner"))
+		}
+	}
+
+	_, err = h.configstoreClient.DeleteOrgInvitation(ctx, orgRef, userRef)
+	if err != nil {
+		return util.NewAPIError(util.KindFromRemoteError(err), errors.Wrapf(err, "failed to delete org invitation"))
+	}
+	return nil
+}
+
+func (h *ActionHandler) acceptOrgInvitation(ctx context.Context, orgInvitation cstypes.OrgInvitation) error {
+	currUserID := common.CurrentUserID(ctx)
+
+	if orgInvitation.UserID != currUserID {
+		return errors.Errorf("user not authorized")
+	}
+
+	isOrgMember, err := h.IsOrgMember(ctx, currUserID, orgInvitation.OrganizationID)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if isOrgMember {
+		return util.NewAPIError(util.ErrForbidden, errors.Errorf("user is just an org member"))
+	}
```

Comment:
If the user is already an org member then there's something wrong in some parts since an invitation should not exists for an org member.